platform/commons/src/main/java/com/peregrine/commons/util/PerUtil.java

```diff
@@ -914,4 +915,68 @@ public boolean doAdd(final Resource resource) {
         @Override
         public boolean doAddChildren(Resource resource) { return true; }
     }
+
+    /**
+     * Extracts the tenant name from the resource
+     * @param resource a resource
+     * @return tenant name on success, and <code>null</code> otherwise
+     */
+    public static String getTenantNameFromResource(final Resource resource) {
+        return (resource != null)
+                ? getTenantNameFromPath(resource.getPath())
```

Comment:
this does not make sense to me. we can check the node names at the different levels directly

Reply:
This method simply delegates to the `getTenantNameFromPath(String)` method. It's quite a bit simpler than iterating the nodes and checking node names., but I will update the implementation per your request.

Reply:
Method refactored. Please review.